src/status_line_manager.ts

```diff
@@ -33,28 +31,38 @@ export class StatusLineManager implements Disposable {
         this.statusBar = window.createStatusBarItem(StatusBarAlignment.Left, -10);
         this.statusBar.show();
         this.disposables.push(this.statusBar, eventBus.on("redraw", this.handleRedraw, this));
-        actions.add("refresh-status-items", (items: string[]) => this.setStatus(items, StatusType.Custom));
-    }
 
-    private setStatus(status: string[], type: StatusType.Custom): void;
-    private setStatus(status: string, type: Omit<StatusType, "Custom">): void;
-    private setStatus(status: string | string[], type: StatusType): void {
-        if (Array.isArray(status)) {
-            this._customStatus = status;
-        } else {
-            switch (type) {
-                case StatusType.Mode:
-                    this._modeText = status;
-                    break;
-                case StatusType.Cmd:
-                    this._cmdText = status;
-                    break;
-                case StatusType.Msg:
-                    this._msgText = status;
-                    break;
+        const refreshNvimStatusLineTimer = setInterval(async () => {
+            let sl = (await this.client.lua(`
+            if vim.o.laststatus == 0 then return "" end
+            return vim.api.nvim_eval_statusline(vim.o.statusline, {}).str
+            `)) as any as string;
```

Comment:
Just FYI: after the next node-client release, the `as any` can be dropped. https://github.com/neovim/node-client/pull/353